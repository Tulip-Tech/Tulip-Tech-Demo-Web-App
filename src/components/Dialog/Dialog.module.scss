@use 'src/styles/variables';
@use 'src/styles/theme';

.dialog {
  width: 100vw;
  max-width: 450px;
  max-height: 90vh;
  padding: 24px;
  overflow-y: auto;
  color: theme.$modal-color;
  background-color: theme.$modal-bg;
  border-radius: 6px;
  -ms-overflow-style: none;
  //noinspection CssUnknownProperty
  scrollbar-width: none;

  &::-webkit-scrollbar {
    display: none;
  }
}
