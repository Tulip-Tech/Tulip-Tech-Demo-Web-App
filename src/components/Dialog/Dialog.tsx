import React from 'react';

import styles from './Dialog.module.scss';

import Modal from '#components/Modal/Modal';
import Slide from '#components/Animation/Slide/Slide';
import ModalCloseButton from '#components/ModalCloseButton/ModalCloseButton';

type Props = {
  open: boolean;
  onClose: () => void;
  children: React.ReactNode;
};

const Dialog: React.FC<Props> = ({ open, onClose, children }: Props) => {
  return (
    <Modal open={open} onClose={onClose} AnimationComponent={Slide}>
      <div className={styles.dialog}>
        <ModalCloseButton onClick={onClose} />
        {children}
      </div>
    </Modal>
  );
};

export default Dialog;
