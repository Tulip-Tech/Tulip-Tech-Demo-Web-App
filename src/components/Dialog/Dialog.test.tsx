import React from 'react';
import { render } from '@testing-library/react';

import Dialog from './Dialog';

describe('<Dialog>', () => {
  test('renders and matches snapshot', () => {
    const { baseElement } = render(
      <>
        <span>Some content</span>
        <Dialog onClose={vi.fn()} open={true}>
          Dialog contents
        </Dialog>
        <span>Some other content</span>
      </>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
