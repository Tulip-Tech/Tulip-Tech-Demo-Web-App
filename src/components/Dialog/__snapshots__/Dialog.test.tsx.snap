// Vitest Snapshot v1

exports[`<Dialog> > renders and matches snapshot 1`] = `
<body
  style="margin-right: 0px; overflow-y: hidden;"
>
  <div>
    <span>
      Some content
    </span>
    <span>
      Some other content
    </span>
  </div>
  <div
    style="transition: opacity 0.3s ease-in-out; opacity: 0;"
  >
    <div
      class="_modal_6c6c55"
    >
      <div
        class="_backdrop_6c6c55"
        data-testid="backdrop"
      />
      <div
        class="_container_6c6c55"
      >
        <div
          style="transition: transform 0.2s ease, opacity 0.2s ease; transform: translate(0, -15px); opacity: 0; z-index: 15;"
        >
          <div
            class="_dialog_00d559"
          >
            <div
              aria-label="close_modal"
              class="_iconButton_0fef65 _modalCloseButton_b92d69"
              role="button"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="_icon_1e9999"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </g>
              </svg>
            </div>
            Dialog contents
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
